File: Labs/Laboratorio1/Pregunta1/Admision.cs
using System;
using System.ComponentModel;

namespace Pregunta1 {
	
	public class Admision {
		private BindingList<FichaEvaluacion> listaFichas;
		private int cantidad_admitidos;
		private int cantidad_postulantes;
		
		public Admision(){
			this.listaFichas = new BindingList<FichaEvaluacion>();
			this.cantidad_admitidos = 0;
			this.cantidad_postulantes = 0;
		}
		
		public void agregarFichaDeEvaluacion(FichaEvaluacion ficha){
			FichaEvaluacion fichaCopia = new FichaEvaluacion(ficha);
			listaFichas.Add(fichaCopia);
			this.cantidad_postulantes++;
			if(ficha.Estado_candidato == Estado.ADMITIDO){
				this.cantidad_admitidos++;
			}
		}
		
		public override string ToString(){
			string reporte = "PROCESO DE ADMISION:";
			reporte += this.cantidad_postulantes + " postulantes, " + this.cantidad_admitidos  + " admitidos\n";
			reporte += "\tLISTA DE ADMITIDOS: \n";
			foreach(FichaEvaluacion ficha in listaFichas){
				if(ficha.Estado_candidato == Estado.ADMITIDO){
					reporte += "\t\t" + ficha.Candidato + "\n";
				}
			}
			return reporte;
		}
	}
}

File: Labs/Lab1-Java/Pregunta1/src/main/java/com/mycompany/pregunta1/Admision.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.pregunta1;

import java.util.ArrayList;

/**
 *
 * @author Usuario
 */
public class Admision {
    private ArrayList<FichaEvaluacion> lista_fichas;
    private int cantidad_admitidos;
    private int cantidad_postulantes;

    public Admision() {
        lista_fichas = new ArrayList<>();
        cantidad_admitidos=0;
        cantidad_postulantes=0;
    }
    
    public void agregarFichaDeEvaluacion(FichaEvaluacion fichaEvaluacion){
        lista_fichas.add(new FichaEvaluacion(fichaEvaluacion));
        if(fichaEvaluacion.getEstado_candidato()==Estado.ADMITIDO){
            cantidad_admitidos++;
        }
        cantidad_postulantes++;
    }
    @Override
    public String toString(){
        String reporte=String.format("PROCESO DE ADMISION: %d postulantes, %d admitidos\n",
                cantidad_postulantes,cantidad_admitidos);
        reporte+="\tLISTA DE ADMITIDOS:\n";
        for(FichaEvaluacion ficha : lista_fichas){
            if(ficha.getEstado_candidato()==Estado.ADMITIDO)
                reporte+="\t\t"+ficha.getCandidato();
        }
        return reporte;
    }
}
